### PR TITLE
[DPE-6188][DPE-5410] Check configuration and credentials for Azure storage

### PR DESCRIPTION
**Pull request by Batalex**
Changes: 
- Dropped support for Python 3.8. We are building and running the charm on Jammy, so we have 3.10 at our disposal. Keeping 3.8 was a pain for dependency resolution (boto3 has constrains on urllib3 depending on the py version)
- Similar to what we did on https://github.com/canonical/spark-history-server-k8s-operator/pull/58, we now create the container and the path upon relating to `azure-storage-integrator`

### CODE REVIEW

File: tests/integration/test_charm_azure.py [hunk 1]
```diff
@@ -92,10 +91,10 @@ async def test_build_and_deploy(ops_test: OpsTest, charm_versions, azure_credent
...
     sleep(10)
     # Setup container
-    setup_azure_container_for_history_server(
-        azure_credentials["container"], azure_credentials["path"]
-    )
-    logger.info("Azure container and path correctly setup!")
+    # setup_azure_container_for_history_server(
+    #     azure_credentials["container"], azure_credentials["path"]
+    # )
+    # logger.info("Azure container and path correctly setup!")
```
Comment (Batalex) [inline tests/integration/test_charm_azure.py:97]: TODO: Remove this commented code as well as the helpers once we confirmed green CI

**Review by welpaolo — APPROVED** [review]
LGTM!

**Review by theoctober19th — APPROVED** [review]
Looks good to me overall, great job! I've left some nitpick suggestions.

File: src/managers/azure_storage.py [hunk 2]
```diff
@@ -5,12 +5,86 @@
...
+    retry,
+    retry_if_result,
+    stop_after_attempt,
+    wait_fixed,
+)
+
 from common.utils import WithLogging
 from core.domain import AzureStorageConnectionInfo
 
+ACCOUNT_URL_TEMPLATE = "https://{storage_account}.blob.core.windows.net/"
```
[the remark this answers is not in the document]
Comment (Batalex) [reply src/managers/azure_storage.py]: Moved to the dataclass in da83be4539d4a85d5c95b2e6c69f5de81c1cd8c0. Thanks, that makes way more sense to have it there

File: src/managers/azure_storage.py [hunk 3]
```diff
@@ -5,12 +5,86 @@
...
+
 
 class AzureStorageManager(WithLogging):
     """Class exposing business logic for interacting with Azure Storage service."""
 
     def __init__(self, config: AzureStorageConnectionInfo):
         self.config = config
+
+    @cached_property
+    def container(self) -> ContainerClient:
```
[the remark this answers is not in the document]
Comment (Batalex) [reply src/managers/azure_storage.py]: Renamed in da83be4539d4a85d5c95b2e6c69f5de81c1cd8c0

**Pull request merged**